### PR TITLE
T5304: Container add volume bind propagation option

### PR DESCRIPTION
**Pull request by sever-sever**
## Change Summary

 Container add volume bind propagation option
```
set container name c1 volume myvlm propagation rshared
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T5304

## How to test
```
set container name c1 allow-host-networks
set container name c1 description 'foof'
set container name c1 image 'busybox'
set container name c1 volume myvlm destination '/tmp'
set container name c1 volume myvlm propagation 'rshared'
set container name c1 volume myvlm source '/tmp'
```
check:
```
vyos@r14# sudo podman inspect -f '{{.Mounts}}' c1
[{bind  /tmp /tmp   [nosuid nodev rbind] true rshared}]
[edit]
vyos@r14# 

```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/container.xml.in [hunk 1]
```diff
@@ -334,6 +334,26 @@
                 </properties>
                 <defaultValue>rw</defaultValue>
               </leafNode>
+              <leafNode name="propagation">
+                <properties>
```
Comment (c-po) [inline interface-definitions/container.xml.in:338]: There are more then those two options available in podman/docker.

```
shared, slave, private, unbindable, rshared, rslave, runbindable, or rprivate(default). See also mount(2).
```

We probably wan't to add them all?

File: interface-definitions/container.xml.in [hunk 2]
```diff
@@ -334,6 +334,26 @@
...
+                  <help>Volume bind propagation</help>
+                  <completionHelp>
+                    <list>rprivate rshared</list>
+                  </completionHelp>
+                  <valueHelp>
+                    <format>rprivate</format>
+                    <description>Volume bind-propagation rprivate</description>
```
Comment (c-po) [inline interface-definitions/container.xml.in]: We should explain what is `rprivate` to the user.


<html>
<body>


Propagation setting | Description
-- | --
shared | Sub-mounts of the original mount are exposed to replica mounts, and sub-mounts of replica mounts are also propagated to the original mount.
slave | similar to a shared mount, but only in one direction. If the original mount exposes a sub-mount, the replica mount can see it. However, if the replica mount exposes a sub-mount, the original mount cannot see it.
private | The mount is private. Sub-mounts within it are not exposed to replica mounts, and sub-mounts of replica mounts are not exposed to the original mount.
rshared | The same as shared, but the propagation also extends to and from mount points nested within any of the original or replica mount points.
rslave | The same as slave, but the propagation also extends to and from mount points nested within any of the original or replica mount points.
rprivate | The default. The same as private, meaning that no mount points anywhere within the original or replica mount points propagate in either direction.


</body>
</html>

**Review by c-po — CHANGES_REQUESTED** [review]

File: interface-definitions/container.xml.in (hunk 2 above)
Comment (sever-sever) [reply interface-definitions/container.xml.in]: Done!

File: interface-definitions/container.xml.in (hunk 1 above)
Comment (sever-sever) [reply interface-definitions/container.xml.in:338]: Done!

**Review by c-po — APPROVED** [review]

**Pull request merged**